src/client/services/apiCaller/index.js

```diff
@@ -37,6 +37,25 @@ const apiCaller = {
     }).then(res => res.json());
   },
 
+  getLayouts(uri, version) {
+    return fetch(`/api/get-layout-history?${qs.stringify({uri, version})}`, {
+      method: 'GET', headers: {
+        'Content-type': 'application/json',
+        'Accept': 'application/json'
+      }
+    }).then(res =>  res.json());
+  },
+  renderPNG(cyJson) {
```

Comment:
Are you sending a request to generate a picture for each layout?

I think it makes sense to send one request for N latest layouts.  e.g return pictures for the 5 latest layouts.

Reply:
No, the function already renders multiple images in one call. Function should be renamed to renderPNGs.